diff --git a/config.sub b/config.sub
index 38f3d037a785..3e8eeeace3ab 100755
--- a/configfsf.sub
+++ b/configfsf.sub
@@ -1751,7 +1751,7 @@ case $os in
	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr* \
-	     | fiwix* | mlibc* )
+ 	     | fiwix* | mlibc* | mos*)
 		;;
	# This one is extra strict with allowed versions
	sco3.2v2 | sco3.2v[4-9]* | sco5v6*)
