diff --git a/./configure b/./configure.new
index 3bf89a6..b17fac8 100755
--- a/./configure
+++ b/./configure.new
@@ -6796,7 +6796,7 @@ echo "${ECHO_T}$cf_cv_ldflags_search_paths_first" >&6
 		fi
 		cf_cv_rm_so_locs=yes
 		;;
-	(linux*|gnu*|k*bsd*-gnu)
+	(linux*|gnu*|k*bsd*-gnu|mos*)
 		if test "$DFT_LWR_MODEL" = "shared" && test -n "$LD_RPATH_OPT" ; then
 			LOCAL_LDFLAGS="${LD_RPATH_OPT}\$(LOCAL_LIBDIR)"
 			LOCAL_LDFLAGS2="$LOCAL_LDFLAGS"
